### PR TITLE
[quant] Input-Weight Equalization - Conv observer support

### PR DESCRIPTION
**Pull request by angelayi**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61350
* #61287
* #61286
* **#61285**

Summary: Modifies observers to support conv layers and tests to make sure that the observers are returning the expected values for conv inputs.

Test Plan: `python test/test_quantization.py TestEqualizeFx.test_input_weight_eq_observer`

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29557041](https://our.internmc.facebook.com/intern/diff/D29557041)

### CODE REVIEW

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29557041).

File: torch/quantization/fx/_equalize.py [hunk 1]
```diff
@@ -61,9 +61,8 @@ def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
         self.equalization_scale = torch.empty(0)
 
     def forward(self, x_orig):
-        # TODO: Allow for convoluational layers
-        if not (x_orig.ndim == 2):
-            raise ValueError("InputEqualizationObserver only supports Linear layers")
+        if not (x_orig.ndim == 2 or x_orig.ndim == 4):
+            raise ValueError("InputEqualizationObserver only supports Linear and Conv layers")
```
Comment (supriyar) [inline torch/quantization/fx/_equalize.py:65]: We probably need to be able to support conv with 1/2/3 dim as well. Currently looks like this only supports conv2d.

File: test/quantization/fx/test_equalize_fx.py [hunk 2]
```diff
@@ -55,44 +55,64 @@
...
+
+        mins = input.copy()
+        maxs = input.copy()
+        for a in axis_list:
+            mins = mins.min(a)
+            maxs = maxs.max(a)
+
+        return (mins, maxs)
+
+    @given(ndim=st.sampled_from((2, 4)),
```
Comment (jerryzh168) [inline test/quantization/fx/test_equalize_fx.py:74]: maybe add tests for 2 to 5 (linear, conv1d, 2d, 3d)

**Review by jerryzh168 — APPROVED** [review]
please add the tests for conv1d and 3d as well

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29557041).

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29557041).

**Pull request closed without merging**